### PR TITLE
Aruha 952 limit token realms

### PR DESCRIPTION
**Pull request by lmontrieux**
ARUHA-952: Limit token realms

## Description
This PR introduces a new authentication mode, REALM. This mode works like the BASIC mode, but additionally, it ensures that a token comes from a realm contained in a list that can be configured.

## Review
- [ ] Tests
- [ ] Documentation
- [x] CHANGELOG

## Deployment Notes
To use this mode, one should set the following configuration items:
- nakadi.oauth2.mode=REALM
- nakadi.oauth2.realm='/realm1', '/realm2', '/realmN'

### CODE REVIEW

**Comment by antban** [comment]
@lmontrieux Have you thought about adding tests like in https://github.com/zalando/nakadi/blob/master/src/test/java/org/zalando/nakadi/security/BasicModeAuthenticationTest.java ?

File: src/main/java/org/zalando/nakadi/config/SecuritySettings.java [hunk 1]
```diff
@@ -12,6 +12,7 @@
     public enum AuthMode {
         OFF,   // no authentication at all
         BASIC, // only checks that the token is valid (has "uid" scope)
+        REALM, // checks that the token is valid and contains at least one required realm
```
Comment (v-stepanov) [inline src/main/java/org/zalando/nakadi/config/SecuritySettings.java]: Why have you decided to make it a 4th auth mode and not just additional pluggable auth. feature?

Comment (v-stepanov) [inline src/main/java/org/zalando/nakadi/config/SecuritySettings.java]: (I don't suggest to change it, I'm just curious)

**Comment by lmontrieux** [comment]
@antban I added a test in 365160e

**Comment by antban** [comment]
:+1:

**Comment by lmontrieux** [comment]
👍

**Comment by v-stepanov** [comment]
👍

**Comment by antban** [comment]
:+1:

**Comment by zrobgar** [comment]
👍

**Comment by v-stepanov** [comment]
👍

File: src/main/resources/application.yml [hunk 2]
```diff
@@ -72,7 +72,7 @@ nakadi:
     adminClientId: adminClientId
     tokenInfoUrl: https://example.com/tokeninfo
     clientId: stups_aruha-event-store-poc
-    realms: '/arealm', '/anotherone'
+    realms: '/arealm, /anotherone'
```
Comment (v-stepanov) [inline src/main/resources/application.yml]: what do you try to achieve? :)

Comment (lmontrieux) [inline src/main/resources/application.yml]: It should be fine now.
I hope.
The tests say it's fine.

**Comment by v-stepanov** [comment]
👍

**Comment by lmontrieux** [comment]
👍

**Comment by lmontrieux** [comment]
👍

**Comment by v-stepanov** [comment]
👍

**Pull request merged**